src/modules/person/persistence/person.repository.ts

```diff
@@ -425,11 +430,28 @@ export class PersonRepository {
             }
         }
 
+        //save old referrer for person-renamed-event before updating the person
+        if (isPersonRenamedEventNecessary) {
+            oldReferrer = this.getReferrer(personEntity);
+            if (!oldReferrer) {
+                const result: Result<string, DomainError> = await this.usernameGenerator.generateUsername(
+                    person.vorname,
+                    person.familienname,
+                );
+                if (!result.ok) {
+                    return result.error;
+                }
+                oldReferrer = result.value;
+            }
+        }
+
         personEntity.assign(mapAggregateToData(person));
         await this.em.persistAndFlush(personEntity);
 
         if (isPersonRenamedEventNecessary) {
-            this.eventService.publish(PersonRenamedEvent.fromPerson(person));
+            this.eventService.publish(PersonRenamedEvent.fromPerson(person, oldReferrer));
+            // wait for privacyIDEA to update the username
+            await new Promise<void>((resolve: () => void) => setTimeout(resolve, 5000));
```

Comment:
This really isn't pretty, but as I understand it this was accepted by the architects?

Reply:
it was the counter idea from the architects in order to make privacyIDEA renaming synchron.